experiments/unbinned_spectrum_dataset/unbinned_datasets.py

```diff
@@ -0,0 +1,291 @@
+import numpy as np
+from astropy.io import fits
+from gammapy.spectrum import (
+    CountsSpectrum,
+    ReflectedRegionsBackgroundMaker,
+    SpectrumDatasetMaker,
+    SpectrumDataset,
+    SpectrumDatasetOnOff,
+)
+from gammapy.utils.regions import list_to_compound_region
+
+
+class UnbinnedSpectrumDatasetMaker(SpectrumDatasetMaker):
+    """Inheriting SpectrumDatasetMaker but replacing "counts" with "events".
```

Comment:
It is possible to avoid overloading here if you leave the on `events` extraction to the `UnbinnedReflectedRegionsBackgroundMaker`, no?  Simply by adding a `make_on_events` there.
Then you can use the standard `SpectrumDatasetMaker`.